src/simulation/atmos_grids.jl

```diff
@@ -0,0 +1,701 @@
+    FS <: Spaces.ExtrudedFiniteDifferenceSpace,
+    I <: Integer,
+    FT <: Real,
+    SR <: Meshes.StretchingRule,
+    T,
+} <: AbstractAtmosGrid
+    center_space::CS
+    face_space::FS
+
+    nh_poly::I
+
+    h_elem::I
+    radius::FT
+    z_elem::I
+    z_max::FT
+
+    z_stretch::SR
+
+    enable_bubble::Bool
+
+    topography::T
+end
+
+function Base.summary(io::IO, grid::SphereGrid)
+    println(io, "Grid type: $(nameof(typeof(grid)))")
+    println(io, "Number of vertical elements: $(grid.z_elem)")
+    println(io, "Height: $(grid.z_max) meters")
+    println(io, "Vertical grid stretching: $(nameof(typeof(grid.z_stretch)))")
+    # Add information about the stretching, if any
+    for field in fieldnames(typeof(grid.z_stretch))
+        println(io, "  with: $(field): $(getproperty(grid.z_stretch, field))")
+    end
+    println(io, "Radius: $(grid.radius) meters")
+    println(io, "Horizontal elements per edge: $(grid.h_elem)")
+    println(io, "  with: $(grid.nh_poly)-degree polynomials")
+    println(
+        io,
+        "  ",
+        grid.enable_bubble ? "with" : "without",
+        ": bubble correction",
+    )
+    if !isnothing(grid.topography)
+        println(io, "  with: $(grid.topography) topography")
+    end
+end
+
+"""
+function VerticallyStretchedSphereGrid(; nh_poly,
+                                         h_elem,
+                                         radius,
+                                         z_elem,
+                                         z_max,
+                                         dz_bottom,
+                                         dz_top,
+                                         topography = nothing,
+                                         topo_smoothing = false,
+                                         enable_bubble = false,
+                                         comms_ctx = ClimaComms.context(),
+                                         float_type = Float64)
+
+Construct an `SphereGrid` for a cubed sphere with columns with non-uniform
+resolution (as prescribed by
+`ClimaCore.Meshes.GeneralizedExponentialStretching`).
+
+Keyword arguments
+=================
+
+- `nh_poly`: Horizontal polynomial degree.
+- `radius`: Radius of the sphere (in meters).
+- `h_elem`: Number of spectral elements per edge.
+- `x_elem`: Number of spectral elements on the x direction.
+- `x_max`: Length of the box (in meters) (`x_min` is 0).
+- `y_elem`: Number of spectral elements on the y direction.
+- `y_max`: Depth of the box (in meters) (`y_min` is 0).
+- `z_elem`: Number of spectral elements on the vertical direction.
+- `dz_bottom`: Resolution at the lower end of the column (in meters).
+- `dz_top`: Resolution at the top end of the column (in meters).
+- `z_max`: Height of the column (in meters).
+- `topography`: Define the surface elevation profile. Provided as a warp function (or nothing).
+- `topo_smoothing`: Whether to order-2 smoothing on the LGL mesh.
```

Comment:
First thing to figure out is what topography we want to support. Currently we have

https://github.com/CliMA/ClimaAtmos.jl/blob/d9788516b6831b2827761f6b1ceeb135d1744612/src/solver/type_getters.jl#L129

- no topography
- analytic cases
  - box/plane: Schar, Agnesi
  - sphere: DCMIP200, mountain from shallow water case (not in ClimaAtmos)

- from data file
  - is preprocessed, with initial filtering applied to the data on the file lat/long grid (`"smoothing_order"`)
    - https://github.com/CliMA/ClimaAtmos.jl/blob/d9788516b6831b2827761f6b1ceeb135d1744612/src/solver/type_getters.jl#L138-L154
  - interpolated to cubed sphere, with additional smoothing applied after via`"topo_smoothing"`
    - https://github.com/CliMA/ClimaAtmos.jl/blob/d9788516b6831b2827761f6b1ceeb135d1744612/src/utils/common_spaces.jl#L100